File: python/interface.py
class Interface():
    """ Do not use all the functions here (dont get board and see where everything is) 
            Feel free to use anything while testing

        You must use - fireShot
        And you can use - isSolved or getShotsFired

    """
    def __init__(self, fileName):
        self.fired = 0
        self.pieceKey = {"A":"Carrier","B":"Battleship","C":"Cruiser","S":"Submarine","D":"Destroyer"}

        self.setBoard(fileName)

    def getBoard(self):
        return self.board

    def setBoard(self, fileName):
        # note how this is one directory back
        textFile = open("..\\"+fileName,"r") #todo dynamic
        fullText = textFile.read()
        textFile.close()
        
        # the reason we have both self.board and self.orginalBoard
        # is because we are marking up self.board as the game goes on
        # when a shot is fired, if it hits a ship we change a C for example
        # to a 1
        self.board = []
        self.orginalBoard = [] # to print at the end board
        for line in fullText.split("\n"):
            self.board.append(line.split())
            self.orginalBoard.append(line.split())
        #todo add check board


    def fireShot(self, x, y):
        self.fired += 1 # increment shots
        shot = self.board[y][x]
        if shot == "0": #missed
            return "0"
        else: #hit
            self.board[y][x] = "1"
            if self.isSolved():
                self.setDone()

            if shot not in str(self.board): #sunk
                return self.pieceKey[shot]
            else: #just hit it
                return "1"

    def getShotsFired(self):
        return self.fired

    def printBoard(self):
        for line in self.board:
            print(" ".join(line))
    def printOriginalBoard(self):
        for line in self.orginalBoard:
            print(" ".join(line))

    def isSolved(self):
        # if there are letters in the board it will fail
        try:
            for row in self.board:
                for char in row:
                    int(char)
            return True
        except:
            return False

    def setDone(self):
        """Done Execution"""
        self.printOriginalBoard()
        print("Turns:", self.getShotsFired())
        self.printBoard()

File: java/Interface.java
import java.io.File;
import java.io.FileReader;
import java.io.IOException;
import java.util.*;
class Interface {
    // todo the orginal board and board are currently the same
    // dont want to spend the time to figure out how to fix it
    
    /** Do not use all the functions here (dont get board and see where everything is) 
            Feel free to use anything while testing

        You must use - fireShot
        And you can use - isSolved or getShotsFired

    */

    public int fired;  
    public Map<String, String> pieceKey;
    public String[][] board;
    public String[][] orginalBoard;

    public Interface(String fileName){
        this.fired = 0;

        // The following is equivalent to Pythons dictionary
        // for example 
        // System.out.println(this.pieceKey.get("A"));
        // will return "Carrie"
        this.pieceKey = new HashMap<String, String>();
        this.pieceKey.put("A", "Carrier");
        this.pieceKey.put("B", "Battleship");
        this.pieceKey.put("C", "Cruiser");
        this.pieceKey.put("S", "Submarine");
        this.pieceKey.put("D", "Destroyer");

        setBoard(fileName);
    }
    public void setBoard(String fileName){
        // read file and add to board array
        String content = null;
        File file = new File("..\\"+fileName); //for ex foo.txt
        try {
           FileReader reader = new FileReader(file);
           char[] chars = new char[(int) file.length()];
           reader.read(chars);
           content = new String(chars);
           reader.close();
        } catch (IOException e) {
           e.printStackTrace();
        }

        // appending to this board and orginal board
        String[][] board = new String[10][10];
        String[][] orginalBoard = new String[10][10];
        String[] lineList = content.split("\n");
        for (int i = 0; i < lineList.length; i++){
            String[] charList = lineList[i].split(" ");

            // for some reason this one has an extra char
            int last = charList.length-1;
            charList[last] = charList[last].substring(0,1);           

            board[i] = charList;
            orginalBoard[i] = charList;
        }


        this.board = board;
        this.orginalBoard = orginalBoard;
    }
    public void printBoard(){
        for (int i = 0; i < this.board.length; i++){
            for (int x = 0; x < this.board[i].length; x++){
                System.out.print(this.board[i][x]);
            }
            System.out.println();
        }
    }
    public void printOrginalBoard(){
        for (int i = 0; i < this.orginalBoard.length; i++){
            for (int x = 0; x < this.orginalBoard[i].length; x++){
                System.out.print(this.orginalBoard[i][x]);
            }
            System.out.println();
        }
    }
    public int getShotsFired(){
        return this.fired;
    }  
    public String fireShot(int x, int y){
        //Zach is waaaay cooler than Dan
        //Dan also needs to learn to comment his code
        //I can't understand half the stuff in this method
        this.fired += 1;
        String shot = this.board[y][x];
        if (shot.equals("0")){
            return "0";
        }
        else{
            this.board[y][x] = "1";
            
            // turn this.board into a string
            String strBoard = "";
            for (int i = 0; i < this.board.length; i++){
                for (int z = 0; z < this.board[i].length; z++){
                    strBoard += this.board[i][z];
                }
            }

            // the shot is not in the board i.e. its sunk
            if (strBoard.indexOf(shot) == -1){
                return this.pieceKey.get(shot);
            }
            else{ // its just a hit
                return "1";
            }
        }
    }
    public boolean isSolved(){
        // if all the characters are numbers the puzzled is solved
        try{
            for (int i = 0; i < this.board.length; i++){
                for (int x = 0; x < this.board[i].length; x++){
                    Integer.parseInt(this.board[i][x]);
                }
            }
            return true;
        }
        catch(NumberFormatException e){
            return false;
        }
    }
    public void setDone(){
        this.printOrginalBoard();
        System.out.println("Turns:"+this.getShotsFired());
        this.printBoard();
    }
}

